dataclients/kubernetes/clusterclient.go

```diff
@@ -374,7 +414,16 @@ func (c *clusterClient) logMissingRouteGroupsOnce() {
 }
 
 func (c *clusterClient) fetchClusterState() (*clusterState, error) {
-	ingresses, err := c.loadIngresses()
+	var (
+		err         error
+		ingressesV1 []*definitions.IngressV1Item
+		ingresses   []*definitions.IngressItem
+	)
+	if c.ingressV1 {
```

Comment:
i am trying to think about a migration path. I thought we would support both variants for a while. How will it work? Wouldn't we end up with a chicken-and-egg situation when considering migrating the resources and skipper to v1?

Reply:
Ok this questions is a miss understanding of Kubernetes.
In Kubernetes you ask as client "please give me resource A in version vN". Kubernetes will translate v1 resources to v1beta1. In kubernetes 1.22 this won't happen anymore and since 1.18 (IIRC) you can already fetch v1 ingress, no matter who created the ingress in which version.

Reply:
got it, thx